bin/compile

```diff
@@ -263,8 +263,7 @@ install_bins() {
     if [[ "$package_manager" == pnpm* ]]; then
       package_manager="$package_manager"
     else
-      warn "Currently pnpm installation is only supported through the 'packageManager' field in package.json"
-      fail
+      package_manager="pnpm@latest"
```

Comment:
Customers often end up with surprises whenever they use defaults. A deploy with no dependency changes might actually install different dependencies if the default pnpm version changes.

What do you think about a warning like:

```bash
warn "pnpm was detected, but the packageManager field in package.json did not define a pnpm version. This build will use `pnpm@latest` by default. We highly suggest setting this field to prevent surprises. You can set this field with `corepack use pnpm@latest` and commit the result."
```

A warning like the above might also help if the customer happens to have a pnpm lockfile, but for some reason had `npm` or `yarn` in `packageManager`.

Reply:
For consistency with npm and Yarn, I chose not to do that.  But I'm all for being more verbose regarding what's happening during the build so I'll add the extra warning message to supplement how this is currently communicated (for pnpm only).

Reply:
@joshwlewis I added a more detailed warning. You can see it in the output if you expanded `functional-test (20)` or `functional-test (22)` and scroll to almost the bottom:
https://github.com/heroku/heroku-buildpack-nodejs/actions/runs/8528738483/job/23363039246?pr=1226